File: banking-app-fronted/src/services/authServices/authService.ts
import api from "../api";
import type {
  LoginRequest,
  LoginResponse,
  RegisterRequest,
  RegisterResponse,
  RefreshTokenRequest,
  RefreshTokenResponse,
  MeResponse,
  ForgotPasswordRequest,
  ForgotPasswordResponse,
  verifyResetCodeRequest,
  verifyResetCodeResponse,
  resetPasswordRequest,
  resetPasswordResponse,
} from "../../types/auth";
import type { ApiResponse } from "../../types/ApiResponse";

export const authService = {
  register: async (data: RegisterRequest): Promise<ApiResponse<RegisterResponse>> => {
    const response: ApiResponse<RegisterResponse> = await api.post(
      "/auth/register",
      data
    );

    return response;
  },

  login: async (data: LoginRequest): Promise<ApiResponse<LoginResponse>> => {
    const response: ApiResponse<LoginResponse> = await api.post(
      "/auth/login",
      data
    );

    return response;
  },

  refresh: async (data: RefreshTokenRequest): Promise<ApiResponse<RefreshTokenResponse>> => {
    const response: ApiResponse<RefreshTokenResponse> = await api.post(
      "/auth/refresh",
      data
    );

    return response;
  },

  me: async (): Promise<ApiResponse<MeResponse>> => {
    const response: ApiResponse<MeResponse> = await api.get(
      "/auth/me"
    );

    return response;
  },

  forgotPassword: async (data: ForgotPasswordRequest): Promise<ApiResponse<ForgotPasswordResponse>> => {
    const response: ApiResponse<ForgotPasswordResponse> = await api.post(
      "/auth/forgot-password",
      data
    );

    return response;
  },

  verifyResetCode: async (data: verifyResetCodeRequest): Promise<ApiResponse<verifyResetCodeResponse>> => {
    const response: ApiResponse<verifyResetCodeResponse> = await api.post(
      "/auth/verify-reset-code",
      data
    );

    return response;
  },

  resetPassword: async (data: resetPasswordRequest): Promise<ApiResponse<resetPasswordResponse>> => {
    const response: ApiResponse<resetPasswordResponse> = await api.post(
      "/auth/reset-password",
      data
    );

    return response;
  }
};

File: banking-app-backend/src/main/java/com/banking/banking_app_backend/auth/service/AuthService.java
package com.banking.banking_app_backend.auth.service;

import com.banking.banking_app_backend.auth.dto.request.*;
import com.banking.banking_app_backend.auth.dto.response.*;
import com.banking.banking_app_backend.common.response.ApiResponse;
import com.banking.banking_app_backend.user.entity.User;
import org.springframework.http.ResponseEntity;

public interface AuthService {

    RegisterResponse register(RegisterRequest registerRequest);

    LoginResponse login(LoginRequest loginRequest);

    RefreshResponse refresh(RefreshRequest refreshRequest);

    MeResponse me(User user);

    ForgotPasswordResponse forgotPassword(ForgotPasswordRequest forgotPasswordRequest);

    VerifyResetCodeResponse verifyResetCode(VerifyResetCodeRequest request);

    ResetPasswordResponse resetPassword(ResetPasswordRequest request);

}
